crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -230,6 +231,31 @@ impl<T: Component + Clone + Eq> State<T> {
         Ok(())
     }
 
+    /// Schedule a state change that replaces the full stack with the given state.
+    /// This will fail if there is a scheduled operation, or if the given `state` matches the current state
+    pub fn set_replace(&mut self, state: T) -> Result<(), StateError> {
```

Comment:
I'm thinking we should simplify naming here:

* set_replace -> replace
* set_next -> next
* set_push -> push
* set_pop -> pop

Reply:
And maybe "next" should be "set"?

"next" could accidentally be interpreted as "queue".

Reply:
I agree with both of those: "next" being interpreted as "push" seems very likely.